### PR TITLE
Upgrade to Jersey 2.27

### PR DESCRIPTION
**Pull request by timcharper**
Pre-requisite for non-blocking responses in Jetty.

- We get rid of `MarathonMediaType.PREFERRED_APPLICATION_JSON` because it specified a qs value that violates the HTTP specification, and Jersey 2 no longer allows it. I have confirmed that the bug described in #1647 does not come back with its removal.

JIRA Issue: MARATHON-8174

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/MarathonExceptionMapper.scala [hunk 1]
```diff
@@ -73,7 +73,7 @@ class MarathonExceptionMapper extends ExceptionMapper[JavaException] {
 
   private def entity(exception: JavaException): JsValue = exception match {
     case e: NotFoundException =>
-      Json.obj("message" -> s"URI not found: ${e.getNotFoundUri.getRawPath}")
+      Json.obj("message" -> "URI not found")
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/MarathonExceptionMapper.scala]: I think you should say which URI was not found.

Comment (timcharper) [reply src/main/scala/mesosphere/marathon/api/MarathonExceptionMapper.scala]: The path is not available on this exception anymore.

Comment (timcharper) [reply src/main/scala/mesosphere/marathon/api/MarathonExceptionMapper.scala]: Also, why do we have to send this to the client? Won't the client know which URL it requested?

Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/MarathonExceptionMapper.scala]: It's easier to debug in the logs. However, if the exception does not have the uri it's fine.

Comment (timcharper) [reply src/main/scala/mesosphere/marathon/api/MarathonExceptionMapper.scala]: Does this affect the logs? My reading is this is what we send to the client.

**Pull request merged**